src/element.c

```diff
@@ -442,19 +431,11 @@ stumpless_set_element_name( struct stumpless_element *element,
     goto fail;
   }
 
-  new_name = copy_cstring_with_length( name, &new_size );
-  if( !new_name ) {
-    goto fail;
-  }
-
-
   lock_element( element );
-  old_name = element->name;
-  element->name = new_name;
-  element->name_length = new_size;
+  snprintf( element->name, sizeof(element->name), "%s", name );
```

Comment:
Again, remove the use of `snprintf` here as well. Use the definition of `stumpless_set_entry_app_name` as a template for what to do here, as the `strlen` can be done outside of the critical section to improve performance.